### PR TITLE
Add support for ingress tls entry

### PR DESCRIPTION
**Pull request by hardillb**
part of #83

Basic TLS ingress entry

## Related Issue(s)

#83

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by dfulgham** [comment]
Should have the tls section refer to a specific secret for the ca certificates.  

I don't know if nginx-ingress can reference/use a configMap (ff-ca-certs), looks like it only supports secrets.  ?

File: kubernetes.js [hunk 1]
```diff
@@ -343,6 +343,14 @@ const createIngress = async (project, options) => {
     localIngress.spec.rules[0].host = url.host
     localIngress.spec.rules[0].http.paths[0].backend.service.name = `${prefix}${project.safeName}`
 
+    if (url.protocol === 'https') {
+        localIngress.spec.tls = [{
+            hosts: [
+                url.host
+            ]
+        }]
```
Comment (dfulgham) [inline kubernetes.js:351]: secretName: "ff-apps-ca-secret"

Or something, to hold the wildcard ca certificate to be used for applications, would need to be documented.

Comment (hardillb) [reply kubernetes.js:352]: `ff-apps-ca-secrets` is not a Secret, it's a collection of CA certs that the instance should trust, so not useful here.

I'm still working out how to solve this nicely, there are 2 possible approaches

1. A single shared wildcard cert/key in a single secret
2. A secret per instance, and something like CertManager issuing new ACME provisioned certs for each.

Both have pros/cons for different environments.

**Comment by hardillb** [comment]
superseded by https://github.com/FlowFuse/driver-k8s/pull/131

**Pull request closed without merging**